CenterForOpenScience/osf.io
# [feature] Add button to remove bad prereg challenge checkouts [OSF-8333]

## Purpose
Checkins don't seem to always occur when a prereg challenge is approved/rejected. At least one source of these was fixed (deleting a prereg submission while under review). Other sources may still exist that have not been identified. To make it easier to handle them, a button has been added to the prereg view that allows bad checkouts to be identified and handled all at once.


## Changes
In addition to the button mention below, Clicking to view approved and rejected prereg challenge items should no longer check out files. 

Add the below button that removes checkouts by prereg admins from rejected or approved prereg nodes. 
![screen shot 2017-08-04 at 4 11 04 pm](https://user-images.githubusercontent.com/1322421/28985361-3368851a-7930-11e7-8921-9935a75ffdb1.png)
![screen shot 2017-08-04 at 4 11 14 pm](https://user-images.githubusercontent.com/1322421/28985363-371c1b86-7930-11e7-9d69-cb361a507472.png)



## Side effects
None known.


## Ticket

https://openscience.atlassian.net/browse/OSF-8333

## QA Notes
1. Three projects: Approved prereg challenge, Unapproved (active prereg submission), Rejected prereg challenge.
2. Add a prereg admin to the project and manually checkout a file.
3. Go to admin app and click on the prereg view and click the "Checkout Checkup" button
4. The count should match the number of projects that have bad prereg checkouts
5. Clicking the button should remove the checkouts from files checked out to the admin from the approved and rejected, but not the unapproved prereg challenge
6. It should not affect any other checkouts but nodes with approved or rejected submissions.
7. Checkouts by other users should not be checked in
8. Viewing approved and rejected preregs on admin should not re-check out the files now
9. Unapproved checkouts should work normally.
